tracer/src/Datadog.Trace/Agent/AgentWriter.cs

```diff
@@ -135,6 +140,48 @@ public void WriteTrace(ArraySegment<Span> trace)
             }
         }
 
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private void LogSpanDebugInformation(TraceContext traceContext, ArraySegment<Span> trace)
+        {
+            foreach (var span in trace)
+            {
+                Log.Debug(
+                    "Span write to agent: [s_id: {SpanId}, p_id: {ParentId}, t_id: {TraceId}] for (Service: {ServiceName}, Resource: {ResourceName}, Operation: {OperationName}, SamplingPriority: {SamplingPriority} Tags: [{Tags}])\nDetails:{ToString}",
+                    new object[] { span.Context.RawSpanId, span.Context.ParentIdInternal, span.Context.RawTraceId, span.ServiceName, span.ResourceName, span.OperationName, traceContext.SamplingPriority, span.Tags, span.ToString() });
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private void LogRootSpanDebugInformation(TraceContext traceContext, Span span)
+        {
+            Log.Debug(
+                    "RootSpan write to agent: [s_id: {SpanId}, p_id: {ParentId}, t_id: {TraceId}] for (Service: {ServiceName}, Resource: {ResourceName}, Operation: {OperationName}, SamplingPriority: {SamplingPriority} Tags: [{Tags}], TraceContext Tags: [{TraceTags}])\nDetails:{ToString}",
+                    new object[] { span.Context.RawSpanId, span.Context.ParentIdInternal, span.Context.RawTraceId, span.ServiceName, span.ResourceName, span.OperationName, traceContext.SamplingPriority, span.Tags, traceContext.Tags, span.ToString() });
+        }
+
+        private string GetSpanInfoForDebugLog(Span span)
```

Comment:
Isn't this very similar to `Span.ToString()`? I think we can consolidate.

Reply:
Actually I think I copy/pasted this over and it actually isn't used here, so can remove

Reply:
Yeah I just copy/pasted it initially but just used the `ToSTring` so removed.